apps/extension/webpack.config.js

```diff
@@ -0,0 +1,146 @@
+  // types declared in prax.d.ts
+  const definitions = {
+    // process.env.NODE_ENV is automatically provided by DefinePlugin
+
+    PRAX: JSON.stringify(PRAX),
+    PRAX_ORIGIN: JSON.stringify(`chrome-extension://${PRAX}`),
+
+    IDB_VERSION: JSON.stringify(IDB_VERSION),
+    USDC_ASSET_ID: JSON.stringify(USDC_ASSET_ID),
+
+    MINIFRONT_URL: JSON.stringify(
+      (argv.mode == 'development' && process.env.MINIFRONT_URL) || MINIFRONT_URL,
+    ),
+
+    DEFAULT_GRPC_URL: JSON.stringify(
+      // you may want https://grpc.testnet-preview.penumbra.zone/
+      (argv.mode == 'development' && process.env.PENUMBRA_NODE_PD_URL) || PENUMBRA_NODE_PD_URL,
+    ),
+  };
+
+  console.log('webpack', { argv, definitions });
+
+  return {
+    entry: {
+      offscreen: path.join(srcDir, 'offscreen.ts'),
+      page: path.join(srcDir, 'page.tsx'),
+      popup: path.join(srcDir, 'popup.tsx'),
+      'injected-connection-manager': path.join(srcDir, 'injected-connection-manager.ts'),
+      'injected-penumbra-global': path.join(srcDir, 'injected-penumbra-global.ts'),
+      'service-worker': path.join(srcDir, 'service-worker.ts'),
+      'wasm-build-action': path.join(srcDir, 'wasm-build-action.ts'),
+    },
+    output: {
+      path: path.join(__dirname, 'dist'),
+      filename: '[name].js',
+    },
+    optimization: {
+      splitChunks: {
+        chunks: chunk =>
+          ![
+            'injected-connection-manager',
+            'injected-penumbra-global',
+            'service-worker',
+            'wasm-build-action',
+          ].includes(chunk.name),
+      },
+    },
+    module: {
+      rules: [
+        { test: /\.wasm/ },
+        {
+          test: /\.tsx?$/,
+          use: 'ts-loader',
+          exclude: /node_modules/,
+        },
+        {
+          test: /\.css$/i,
+          use: [
+            'style-loader',
+            'css-loader',
+            {
+              loader: 'postcss-loader',
+              options: {
+                postcssOptions: {
+                  ident: 'postcss',
+                  plugins: ['tailwindcss', 'autoprefixer'],
+                },
+              },
+            },
+          ],
+        },
+        {
+          test: /\.mp4$/,
+          type: 'asset/resource',
+          generator: {
+            filename: 'videos/[hash][ext][query]',
+          },
+        },
+      ],
+    },
+    resolve: {
+      extensions: ['.ts', '.tsx', '.js'],
+      alias: {
+        '@ui': path.resolve(__dirname, '../../packages/ui'),
+      },
+    },
+    plugins: [
+      new webpack.CleanPlugin(),
+      new webpack.ProvidePlugin({
+        // Required by the `bip39` library
+        Buffer: ['buffer', 'Buffer'],
+      }),
+      new webpack.IgnorePlugin({
+        // Not required by the `bip39` library, but very nice
+        checkResource(resource) {
+          return /.*\/wordlists\/(?!english).*\.json/.test(resource);
+        },
+      }),
+      new webpack.DefinePlugin(definitions),
```

Comment:
We could just do the prod/dev diff here:

```typescript
new webpack.DefinePlugin(argv.mode == 'prod' ? prodDefinitions : devDefinitions)
```

Reply:
removed any prod/dev switching